cmd/kubeadm/app/cmd/reset.go

```diff
@@ -266,26 +267,17 @@ func removeContainers(execer utilsexec.Interface, criSocketPath string) error {
 
 // cleanDir removes everything in a directory, but not the directory itself
 func cleanDir(filePath string) error {
-	// If the directory doesn't even exist there's nothing to do, and we do
-	// not consider this an error
-	if _, err := os.Stat(filePath); os.IsNotExist(err) {
-		return nil
-	}
-
-	d, err := os.Open(filePath)
-	if err != nil {
-		return err
-	}
-	defer d.Close()
-	names, err := d.Readdirnames(-1)
+	names, err := ioutil.ReadDir(filePath)
```

Comment:
if the directory does not exist ReadDir will throw an error, see:
https://play.golang.org/p/x6JlQivxSVY

cleanDir should return without error in such cases.

Reply:
i don't think we have to modify cleanDir as it does not need optimization.

Reply:
+1